<p>When handling a caught exception, two mandatory informations should be either logged, or present in a rethrown exception:</p>

<ul>
  <li>Some context to ease the reproduction of the issue.</li>
  <li>The original's exception, for its message and stack trace.</li>
</ul>

<p>The following code snippet illustrates this rule:</p>

<pre>
try { /* ... */ } catch (Exception e) { LOGGER.info("context"); }         // Non-Compliant - exception is lost
try { /* ... */ } catch (Exception e) { LOGGER.info(e); }                 // Non-Compliant - context is required
try { /* ... */ } catch (Exception e) { LOGGER.info(e.getMessage()); }    // Non-Compliant - exception is lost (only message is preserved )

try { /* ... */ } catch (Exception e) { LOGGER.info("context", e); }      // Compliant

try {
  /* ... */
} catch (Exception e) {                                                   // Non-Compliant - exception is lost
  throw new RuntimeException("context");
}

try {
  /* ... */
} catch (Exception e) {                                                   // Compliant
  throw new RuntimeException("context", e);
}
</pre>
